### ISSUE TITLE
Support tree-shaking by adding a es2015 build target

### ISSUE THREAD

**Issue by swese44** [Microsoft/YamUI#274]
Fabric has multiple build targets, we could do something similar. For reference:

Here's the PR which introduced the es2015 build: https://github.com/OfficeDev/office-ui-fabric-react/pull/3260

Here's the `ts` build task which has multiple targets:
https://github.com/OfficeDev/office-ui-fabric-react/blob/72b103d989212d8c3e7976e6dfc285c17356d9be/scripts/tasks/ts.js

**Comment by swese44** [comment]
UPDATE

we've learned that the underlying Fabric library will soon release 6.x, replacing its default `/lib` export directory with the previously-named /`lib-es2015` which uses es6 modules. instead of adding an alternate build target in YamUI, we've decided to follow suit and export es6 modules only to `/dist`.

apps consuming YamUI which use webpack 3.x will need to update their webpack build config accordingly to transpile YamUI's es6 modules. apps on webpack 4.x will be able to benefit from tree-shaking.

fyi @a-tarasyuk
